src/NUnitFramework/framework/Api/DefaultTestAssemblyBuilder.cs

```diff
@@ -81,42 +81,46 @@ public ITest Build(Assembly assembly, IDictionary<string, object> options)
 #endif
 
             string assemblyPath = AssemblyHelper.GetAssemblyPath(assembly);
-            return Build(assembly, assemblyPath, options);
+            string assemblyName = assemblyPath.IndexOfAny(Path.GetInvalidPathChars()) == -1
```

Comment:
I'm almost happier if we specially recognize `<Unknown>` inside `AssemblyHelper.GetAssemblyPath` and return null.

If there are invalid path characters for any other reason, wouldn't we'd want to know about it right away and not just treat it as though it was the result of a call to `Assembly.ManifestModule.FullyQualifiedName` returning its documented `<Unknown>` value?

Reply:
Ah yes, that makes sense since <unknown> is a special case. I'll make the change to only handle <unknown>.